packages/ember-glimmer/lib/component.js

```diff
@@ -96,26 +95,453 @@ export const BOUNDS = symbol('BOUNDS');
+    awesome: true
+  });
+  ```
+
+  Will result in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view so-very-cool"></div>
+  ```
+
+  Boolean value class name bindings whose property names are in a
+  camelCase-style format will be converted to a dasherized format:
+
+  ```app/components/my-widget.js
+  import Ember from 'ember';
+
+  export default Ember.Component.extend({
+    classNameBindings: ['isUrgent'],
+    isUrgent: true
+  });
+  ```
+
+  Will result in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view is-urgent"></div>
+  ```
+
+  Class name bindings can also refer to object values that are found by
+  traversing a path relative to the component itself:
+
+  ```app/components/my-widget.js
+  import Ember from 'ember';
+
+  export default Ember.Component.extend({
+    classNameBindings: ['messages.empty']
+    messages: Ember.Object.create({
+      empty: true
+    })
+  });
+  ```
+
+  Will result in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view empty"></div>
+  ```
+
+  If you want to add a class name for a property which evaluates to true and
+  and a different class name if it evaluates to false, you can pass a binding
+  like this:
+
+  ```app/components/my-widget.js
+  import Ember from 'ember';
+
+  export default Ember.Component.extend({
+    classNameBindings: ['isEnabled:enabled:disabled']
+    isEnabled: true
+  });
+  ```
+
+  Will result in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view enabled"></div>
+  ```
+
+  When isEnabled is `false`, the resulting HTML representation looks like
+  this:
+
+  ```html
+  <div id="ember1" class="ember-view disabled"></div>
+  ```
+
+  This syntax offers the convenience to add a class if a property is `false`:
+
+  ```app/components/my-widget.js
+  import Ember from 'ember';
+
+  // Applies no class when isEnabled is true and class 'disabled' when isEnabled is false
+  export default Ember.Component.extend({
+    classNameBindings: ['isEnabled::disabled']
+    isEnabled: true
+  });
+  ```
+
+  Will result in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view"></div>
+  ```
+
+  When the `isEnabled` property on the component is set to `false`, it will result
+  in component instances with an HTML representation of:
+
+  ```html
+  <div id="ember1" class="ember-view disabled"></div>
+  ```
+
+  Updates to the value of a class name binding will result in automatic
+  update of the  HTML `class` attribute in the component's rendered HTML
+  representation. If the value becomes `false` or `undefined` the class name
+  will be removed.
+  Both `classNames` and `classNameBindings` are concatenated properties. See
+  [Ember.Object](/api/classes/Ember.Object.html) documentation for more
```

Comment:
@acorncom we should figure out a way to automate to whatever we choose, I think.